src/sknnr/_weighted_trees.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from typing import Callable, Literal
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from ._base import TransformedKNeighborsRegressor, YFitMixin
+from .transformers import RFNodeTransformer
+
+TREE_NODE_TRANSFORMER = RFNodeTransformer
+
+
+class WeightedHammingDistanceMetric:
+    """
+    Distance metric to calculate Hamming distance based on weights.
+    Setting of weights is only supported after initialization.
```

Comment:
It might be worth mentioning explicitly that this works as a callable pairwise distance metric, and that it uses uniform weighting by default.

Reply:
Let me know if I hit the mark here.  I basically copied you verbatim 😉